tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -85,6 +96,56 @@ def test_write(self):
                                   topic=self.topic,
                                   partition_idx=0,
                                   count=6)
-
         self.start_consumer()
         self.run_validation()
+
+
+class EndToEndShadowIndexingTestWithDisruptions(EndToEndShadowIndexingBase):
+    def _build_redpanda_instance(self):
```

Comment:
Polymorphism is a useful thing but it has a cost (jumping between classes to understand the context) so when we may get away without it - it's better to avoid it. In this case we need _build_redpanda_instance only to pass extra config via extra_rp_conf. You can do the same just by passing extra_rp_conf via constructor like we do in the other tests inherited from RedpandaTest.

Reply:
thanks for the suggestion, I have made changes as suggested, extra_rp_conf now passed via constructor